crates/bin/pindexer/src/dex_ex/mod.rs

```diff
@@ -1150,10 +1151,10 @@ impl Component {
                 let num_swaps = filtered_swaps.len() as i32;
 
                 batch_swap_summaries.push(BatchSwapSummary {
-                    asset_start: asset_start.to_bytes().to_vec(),
-                    asset_end: asset_end.to_bytes().to_vec(),
-                    input: BigDecimal::from(input.value()),
-                    output: BigDecimal::from(output.value()),
+                    asset_start: hex::encode(asset_start.to_bytes()),
+                    asset_end: hex::encode(asset_end.to_bytes()),
```

Comment:
If you accept the suggestion to use domain types rather than Strings, then we no longer need the `hex::encode` call. If necessary, we can call `.encode_to_vec()` on domain types, but there should be sufficient serialization handlers already defined, so that you can just put the direct values in here: `asset_start: asset_start`, or just `asset_start,`.